lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1103,6 +1103,17 @@ sub custom_module {
     $self->account_result_size('custom module ' . $module->test->name, $parser->write_output($self->result_dir));
 }
 
+sub job_module {
+    my ($self, $name) = @_;
+    return $self->result_source->schema->resultset('JobModules')->search({job_id => $self->id, name => $name})->first;
+}
+
+sub job_modules {
```

Comment:
We have already `$job->modules` due to the `has_many` relation. What makes this function special is that the job is pre-fetched. Maybe is makes sense to do this pre-fetching in the `has_many` relation by default? Otherwise I'd call this `modules_with_job_prefetched` to be more explicit about the particularity of that function.

Reply:
This one is not so simple, while i agree about the name i'm unsure about the performance implications of just pre-fetching all the time. Given that the pre-fetch change itself is still very fresh (#3126), i'd prefer to keep the method for now and maybe experiment with alternative solutions later on.

Reply:
DBIx has related object cache but it doesn't fill it when you load a list of `has_many` children. Ideally, `$job->modules` should simply put `$job` into the cache for all module objects and the prefetching could be dropped altogether.

Reply:
Specifically, the related job cache is stored in `$module->{_relationship_data}->{job}`.